bin/deployables

```diff
@@ -83,600 +99,829 @@ region=${DEPLOY_AWS_REGION}
+    $lambda_function_arn
 }
 
-function deploy_lambda_event {
-    if [[ -z "${DEPLOY_LAMBDA_EVENT_RULE}" ]] ; then
-        echo "Info: No DEPLOY_LAMBDA_EVENT_RULE, not deploying event."
-        return
-    fi
+function lambda_deploy_event () {
+  if [[ -z "${DEPLOY_LAMBDA_EVENT_RULE}" ]] ; then
+    echo "Info: No DEPLOY_LAMBDA_EVENT_RULE, not deploying event."
+    return
+  fi
 
-    if [[ -z "${DEPLOY_LAMBDA_TARGET_INPUT_JSON}" ]] ; then
-        echo "Info: No DEPLOY_LAMBDA_TARGET_INPUT_JSON, not deploying event."
-        return
-    fi
+  if [[ -z "${DEPLOY_LAMBDA_TARGET_INPUT_JSON}" ]] ; then
+    echo "Info: No DEPLOY_LAMBDA_TARGET_INPUT_JSON, not deploying event."
+    return
+  fi
 
-    aws_configure
+  aws_configure
 
-    lambda_event_basename="${1:-$DEPLOY_LAMBDA_EVENT_BASENAME}"
-    lambda_function_arn="${2:-$DEPLOY_LAMBDA_FUNCTION_ARN}"
-    event_rule_name="${lambda_event_basename}_rule"
-    permission_statement_id="${lambda_event_basename}_permission"
+  lambda_event_basename="${1:-$DEPLOY_LAMBDA_EVENT_BASENAME}"
+  lambda_function_arn="${2:-$DEPLOY_LAMBDA_FUNCTION_ARN}"
+  event_rule_name="${lambda_event_basename}_rule"
+  permission_statement_id="${lambda_event_basename}_permission"
 
-    # this json is a sub-json to something else, so we need to escape the quotes because
-    # the aws cli is stupid
-    input_json_escaped=${DEPLOY_LAMBDA_TARGET_INPUT_JSON//\"/\\\"}
+  # this json is a sub-json to something else, so we need to escape the quotes because
+  # the aws cli is stupid
+  input_json_escaped=${DEPLOY_LAMBDA_TARGET_INPUT_JSON//\"/\\\"}
 
-    if [ -z "${lambda_function_arn}" ] ; then
-        echo "Error: DEPLOY_LAMBDA_FUNCTION_ARN or passing a 'lambda_function_arn' is required for deploy_lambda_event."
-        exit 1
-    fi
+  if [[ -z "${lambda_function_arn}" ]] ; then
+    echo "Error: DEPLOY_LAMBDA_FUNCTION_ARN or passing a 'lambda_function_arn' is required for deploy_lambda_event."
+    exit 1
+  fi
 
-    if [[ -z "${lambda_event_basename}" ]] ; then
-        echo "Error: DEPLOY_LAMBDA_EVENT_BASENAME or passing the basename is required for deploy_lambda_event."
-        exit 1
-    fi
+  if [[ -z "${lambda_event_basename}" ]] ; then
+    echo "Error: DEPLOY_LAMBDA_EVENT_BASENAME or passing the basename is required for deploy_lambda_event."
+    exit 1
+  fi
 
 
-    # create or update a cloudwatch event rule
-    event_rule_arn=$( \
-        aws events put-rule \
-            --profile "${DEPLOY_AWS_PROFILE}" \
-            --name "${event_rule_name}" \
-            --schedule-expression "${DEPLOY_LAMBDA_EVENT_RULE}" \
-            | $JQ ".RuleArn"
-    )
+  # create or update a cloudwatch event rule
+  event_rule_arn=$( \
+    aws events put-rule \
+    --profile "${DEPLOY_AWS_PROFILE}" \
+    --name "${event_rule_name}" \
+    --schedule-expression "${DEPLOY_LAMBDA_EVENT_RULE}" \
+    | $JQ ".RuleArn"
+  )
+
+  # NB this can fail if the permission doesn't exist, this is ok.
```

Comment:
Does the deploy script fail if this call fails? Or does AWS not return an error code when this call fails?

Reply:
I'm not sure, Jeff wrote this, but I believe AWS just fails silently here

Reply:
Cool, just checking